src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -1058,31 +1322,126 @@ def _set_up_defaults(
     resource_group: "ResourceGroup",
     env: "ContainerAppEnvironment",
     app: "ContainerApp",
+    custom_location: "CustomLocation",
+    extension: "Extension"
 ):
     # If no RG passed in and a singular app exists with the same name, get its env and rg
-    _get_app_env_and_group(cmd, name, resource_group, env, location)
+    _get_app_env_and_group(cmd, name, resource_group, env, location, custom_location)
 
     # If no env passed in (and not creating a new RG), then try getting an env by location / log analytics ID
     _get_env_and_group_from_log_analytics(
         cmd, resource_group_name, env, resource_group, logs_customer_id, location
     )
-
-    # try to set RG name by env name
-    if env.name and not resource_group.name:
+    # Find managed env with env name, if found, use it.
+    # Try to set RG name by env name
```

Comment:
you will try to find managed one first????

Reply:
`if not env.is_connected_environment()` has two options:
- env resource type is mangedEnvironment
- env resource type is None